var/spack/repos/builtin/packages/bohrium/package.py

```diff
@@ -0,0 +1,170 @@
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Bohrium(CMakePackage):
+    """Library for automatic acceleration of array operations"""
+
+    homepage    = "http://bh107.org"
+    url         = "https://github.com/bh107/bohrium/archive/v0.8.9.tar.gz"
+    maintainers = ['mfherbst']
+
+    #
+    # Versions
+    #
+    version("develop", git="https://github.com/bh107/bohrium.git",
+            branch="master")
+
+    #
+    # Variants
+    #
+    variant("cuda", default=True,
+            description="Build with CUDA code generator")
+    variant("opencl", default=True,
+            description="Build with OpenCL code generator")
+    variant("openmp", default=True,
+            description="Build with OpenMP code generator")
+
+    variant("node", default=True,
+            description="Build the node vector engine manager")
+    variant("proxy", default=True,
+            description="Build the proxy vector engine manager")
+    variant("python", default=True,
+            description="Build the numpy-like bridge "
+            "to enable use from python")
+    variant("cbridge", default=True,
+            description="Build the bridge interface towards plain C")
+
+    variant("blas", default=True,
+            description="Build with BLAS extension methods")
+    variant("lapack", default=True,
+            description="Build with LAPACK extension methods")
+
+    #
+    # Conflicts and extensions
+    #
+    conflicts("%intel")
+    conflicts("%clang@:3.5")
+    extends("python", when="+python")
+
+    #
+    # Dependencies
+    #
+    depends_on("cmake@2.8:", type="build")
+
+    depends_on("boost+system+serialization+filesystem+regex")
+
+    depends_on("cuda", when="+cuda")
+    depends_on("opencl", when="+opencl")
+
+    # NOTE The lapacke interface and hence netlib-lapack
+    #      is the strictly required lapack provider
+    #      for bohrium right now.
+    depends_on("netlib-lapack+lapacke", when="+lapack")
+    depends_on("blas", when="+blas")
+
+    depends_on("python", type="build", when="~python")
+    depends_on("python", when="+python")
+    depends_on("py-numpy", when="+python")
+    depends_on("swig", type="build", when="+python")
+    depends_on("py-cython", type="build", when="+python")
+
+    depends_on("zlib", when="+proxy")
+
+    #
+    # Settings and cmake cache
+    #
+    def cmake_args(self):
+        spec = self.spec
+
+        # Checks for consistency:
+        if "+node" not in spec and "+proxy" not in spec:
```

Comment:
instead define a conflict outside of `cmake_args`:
```
conflicts('~node~proxy')
```

Reply:
Cool! I wondered whether that worked, but could not find a note about that in the documentation. Much more elegant.

Reply:
it also throws error during conretization, so you get it before building any dependencies.